File: packages/ink-swift/Tests/InkGravesTests/GeometryTests.swift
import InkCore
import Testing

@Suite struct GeometryTests {
    @Test func foldsOffsetsIntoStrokesWithYFlip() {
        let offsets = [
            StrokeOffset(dx: 1, dy: 1, eos: false),
            StrokeOffset(dx: 1, dy: -1, eos: true),
            StrokeOffset(dx: 2, dy: 0, eos: false),
        ]
        let strokes = offsetsToLine(offsets)
        #expect(strokes == [
            InkStroke(points: [SIMD2(1, -1), SIMD2(2, 0)]),
            InkStroke(points: [SIMD2(4, 0)]),
        ])
        #expect(lineBounds(strokes) == Bounds(minX: 1, minY: -1, maxX: 4, maxY: 0))
    }

    @Test func emptyLineHasNoBounds() {
        #expect(offsetsToLine([]).isEmpty)
        #expect(lineBounds([]) == nil)
    }
}

File: packages/ink-kotlin/ink-core/src/test/kotlin/com/trylonghand/ink/core/GeometryTests.kt
package com.trylonghand.ink.core

import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertNull
import kotlin.test.assertTrue

class GeometryTests {
    @Test
    fun foldsOffsetsIntoStrokesWithYFlip() {
        val offsets = listOf(
            StrokeOffset(dx = 1.0, dy = 1.0, eos = false),
            StrokeOffset(dx = 1.0, dy = -1.0, eos = true),
            StrokeOffset(dx = 2.0, dy = 0.0, eos = false),
        )
        val strokes = offsetsToLine(offsets)
        assertEquals(
            listOf(
                InkStroke(listOf(Vec2(1.0, -1.0), Vec2(2.0, 0.0))),
                InkStroke(listOf(Vec2(4.0, 0.0))),
            ),
            strokes,
        )
        assertEquals(Bounds(minX = 1.0, minY = -1.0, maxX = 4.0, maxY = 0.0), lineBounds(strokes))
    }

    @Test
    fun emptyLineHasNoBounds() {
        assertTrue(offsetsToLine(emptyList()).isEmpty())
        assertNull(lineBounds(emptyList()))
    }
}
